src/events/upgrade.py

```diff
@@ -33,7 +32,7 @@ class ZooKeeperDependencyModel(BaseModel):
     service: DependencyModel
 
 
-class ZooKeeperUpgrade(DataUpgrade):
+class ZooKeeperEvents(DataUpgrade):
```

Comment:
**todo** we could also re-write some of the charm logic here using the new abstraction, e.g. not use SNAP class but using the Workload class

Reply:
Agreed, this section has had the least attention and probably needs an update, will do!